app/src/androidTest/java/com/github/wnder/GuessLocationTourModeActivityTest.java

```diff
@@ -0,0 +1,191 @@
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static com.github.wnder.TourDatabaseTest.deleteTestTour;
+import static org.hamcrest.Matchers.not;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@HiltAndroidTest
+@UninstallModules({PicturesModule.class})
+public class GuessLocationTourModeActivityTest {
+    //Intent with extras that the activity will start with
+    private static Intent intent;
+
+    @BindValue //For the creation of the tour in the database
+    public static PicturesDatabase picturesDatabase = Mockito.mock(PicturesDatabase.class);
+
+    private final HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION);
+
+    @Rule
+    public RuleChain testRule = RuleChain.outerRule(hiltRule)
+            .around(new ActivityScenarioRule<>(intent));
+
+    //For activities that we did ourself otherwise need to use mockito
+    @Before
+    public void setUp() {
+        Intents.init();
+    }
+
+    @After
+    public void tearDown(){
+        Intents.release();
+    }
+
+    @BeforeClass
+    public static void beforeAll() throws ExecutionException, InterruptedException{
+
+        FirebaseTourDatabase tdb = new FirebaseTourDatabase(ApplicationProvider.getApplicationContext());
+        FirebasePicturesDatabase db = new FirebasePicturesDatabase(ApplicationProvider.getApplicationContext());
+
+        Location defaultPicLoc = new Location("");
+        defaultPicLoc.setLatitude(90);
+        defaultPicLoc.setLongitude(0);
+
+        String userName = "testUser";
+        String firstUniqueId = "first" + userName + Calendar.getInstance().getTimeInMillis();
+
+        CompletableFuture<Void> uploadFirstPic = db.uploadPicture(firstUniqueId, new UploadInfo(userName, defaultPicLoc, Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag)));
+
+        uploadFirstPic.get();
+
+        String tourName = "testTour";
+        String tourUniqueId = tdb.generateTourUniqueId(tourName);
+
+        List<String> pictures = new ArrayList<>();
+        pictures.add(firstUniqueId);
+
+        CompletableFuture<Void> uploadTour = tdb.uploadTour(tourUniqueId, tourName, pictures);
+
+        try{
+            uploadTour.get();
+        } catch(Exception e){
+            e.printStackTrace();
+        }

```

Comment:
The exception should be added to the method signature and try/catch should be removed so that the test always fails if an exception is thrown.

Reply:
Done